spec/ddtrace/sync_writer_spec.rb

```diff
@@ -10,4 +10,54 @@
     subject(:runtime_metrics) { sync_writer.runtime_metrics }
     it { is_expected.to be_a_kind_of(Datadog::Runtime::Metrics) }
   end
+
+  describe '#write' do
+    subject(:write) { sync_writer.write(trace, services) }
+    let(:trace) { get_test_traces(1).first }
+    let(:services) { nil }
+
+    context 'with report hostname' do
+      context 'enabled' do
+        around do |example|
+          Datadog.configuration.report_hostname = Datadog.configuration.report_hostname.tap do
+            Datadog.configuration.report_hostname = true
+            example.run
+          end
+        end
+
+        it do
+          expect(sync_writer.transport).to receive(:send) do |_type, traces|
+            root_span = traces.first.first
+            expect(root_span.get_tag(Datadog::Ext::NET::TAG_HOSTNAME)).to be_a_kind_of(String)
```

Comment:
We could do `allow()` on `::Socket.gethostname` if we wanted to assert on a specific value.

Reply:
Hmmm yeah, but for the sake of test practice and trying to keep these from becoming brittle, I don't want tests to assume the internals about how things works of which it shouldn't. `SyncWriter` should not know how hostname is resolved, it should only assert the publicly visible side effects.

Perhaps a better way of doing this would be `allow(Datadog::Runtime::Socket).to receive(:hostname)`, which would maintain this obfuscation but allow us to assert the value from that is set on the tag.

Reply:
yeah, the idea being that if we mock around `::Socket.gethostname` we don't care what `Socket.gethostname` does to resolve the hostname, but we can assert that `Datadog::Runtime::Socket.hostname` does what we expect with it.

Today.. nothing, but maybe we need to `#trim!` or in the future we look do some post processing on it, etc.

Mocking over `::Socket.gethostname`'s return value should allow us to assert the value of the tag without making the tests brittle since it the value will be constant.

Reply:
`Datadog::Runtime::Socket` has its own unit tests in which we could mock/stub calls to `Socket`. This should not happen in `SyncWriter` though, as its multiple levels of indirection away.

We can stub the call to `Datadog::Runtime::Socket#hostname` here though; that would be appropriate.

Reply:
Sure, that is fair.